sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -247,4 +248,21 @@ class UDFSuite extends QueryTest with SharedSQLContext {
       sql("SELECT tmp.t.* FROM (SELECT testDataFunc(a, b) AS t from testData2) tmp").toDF(),
       testData2)
   }
+
+  test("Column aliases are ignored in callUDF while using struct()") {
```

Comment:
This test case passes without any of the other changes in this PR.  Please check to make sure that you are actually testing what you think you are testing.